test/with_dummyserver/test_https.py

```diff
@@ -465,29 +465,33 @@ def test_assert_fingerprint_sha256(self):
             https_pool.request("GET", "/")
 
     def test_assert_invalid_fingerprint(self):
+        def _test_request(pool):
+            with pytest.raises(MaxRetryError) as cm:
+                pool.request("GET", "/", retries=0)
+            assert isinstance(cm.value.reason, SSLError)
+            return cm.value.reason
+
         with HTTPSConnectionPool(
-            "127.0.0.1", self.port, cert_reqs="CERT_REQUIRED", ca_certs=DEFAULT_CA
+            self.host, self.port, cert_reqs="CERT_REQUIRED", ca_certs=DEFAULT_CA
         ) as https_pool:
+
             https_pool.assert_fingerprint = (
                 "AA:AA:AA:AA:AA:AAAA:AA:AAAA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA"
             )
-
-            def _test_request(pool):
-                with pytest.raises(MaxRetryError) as cm:
-                    pool.request("GET", "/", retries=0)
-                assert isinstance(cm.value.reason, SSLError)
-
-            _test_request(https_pool)
+            e = _test_request(https_pool)
+            assert "Fingerprints did not match." in str(e)
             https_pool._get_conn()
```

Comment:
Why are these calls to `_get_conn()` necessary?

Reply:
They were added in https://github.com/urllib3/urllib3/commit/e6adf684591b71c562a17b05b0df374cd39d76be.
I don't see why they are necessary, removed them for now.